src/data/rat/basic.lean

```diff
@@ -744,6 +744,26 @@ begin
     coe_nat_div_self]
 end
 
+lemma inv_coe_nat_num {a : ℕ} (ha0 : 0 < a) : (a : ℚ)⁻¹.num = 1 :=
+begin
+  rw [rat.inv_def', rat.coe_nat_num, rat.coe_nat_denom],
+  suffices : (((1 : ℤ) : ℚ) / (a : ℤ)).num = 1,
+    exact_mod_cast this,
+  apply num_div_eq_of_coprime,
+  { assumption_mod_cast },
+  { simp only [nat.coprime_one_left_iff, int.nat_abs_one] }
+end
```

Comment:
I think this proof is simpler if you prove it for `int` first:
```
lemma inv_coe_int_num {a : ℤ} (ha0 : 0 < a) : (a : ℚ)⁻¹.num = 1 :=
begin
  rw [rat.inv_def', rat.coe_int_num, rat.coe_int_denom, nat.cast_one, ←int.cast_one],
  apply num_div_eq_of_coprime ha0,
  rw int.nat_abs_one,
  exact nat.coprime_one_left _,
end

lemma inv_coe_nat_num {a : ℕ} (ha0 : 0 < a) : (a : ℚ)⁻¹.num = 1 :=
inv_coe_int_num (by exact_mod_cast ha0 : 0 < (a : ℤ))
```
Can you do the same for the lemma below?

Reply:
Yeah, thanks for the feedback!  I updated your golf and added `inv_coe_int_denom`.  The nat version of the proof uses and additional `exact_mod_cast`, because of casting the denom, which is a nat.